store/sqlstore/upgrade.go

```diff
@@ -106,28 +147,11 @@ func UpgradeDatabase(sqlStore SqlStore) {
 	UpgradeDatabaseToVersion510(sqlStore)
 	UpgradeDatabaseToVersion511(sqlStore)
 
-	// If the SchemaVersion is empty this this is the first time it has ran
-	// so lets set it to the current version.
-	if sqlStore.GetCurrentSchemaVersion() == "" {
-		if result := <-sqlStore.System().SaveOrUpdate(&model.System{Name: "Version", Value: model.CurrentVersion}); result.Err != nil {
-			mlog.Critical(result.Err.Error())
-			time.Sleep(time.Second)
-			os.Exit(EXIT_VERSION_SAVE_MISSING)
-		}
-
-		mlog.Info(fmt.Sprintf("The database schema has been set to version %v", model.CurrentVersion))
-	}
-
-	// If we're not on the current version then it's too old to be upgraded
-	if sqlStore.GetCurrentSchemaVersion() != model.CurrentVersion {
-		mlog.Critical(fmt.Sprintf("Database schema version %v is no longer supported. This Mattermost server supports automatic upgrades from schema version %v through schema version %v. Downgrades are not supported. Please manually upgrade to at least version %v before continuing", sqlStore.GetCurrentSchemaVersion(), OLDEST_SUPPORTED_VERSION, model.CurrentVersion, OLDEST_SUPPORTED_VERSION))
-		time.Sleep(time.Second)
-		os.Exit(EXIT_TOO_OLD)
-	}
+	return nil
 }
 
 func saveSchemaVersion(sqlStore SqlStore, version string) {
-	if result := <-sqlStore.System().Update(&model.System{Name: "Version", Value: version}); result.Err != nil {
+	if result := <-sqlStore.System().SaveOrUpdate(&model.System{Name: "Version", Value: version}); result.Err != nil {
```

Comment:
The existing unit tests weren't actually writing the version, since the row didn't exist. There's no circumstances in which we would want this behaviour, so I'm inserting or updating all the time now.